pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -434,58 +443,55 @@ func (sb stageBuilder) build() Stage {
 // may prove to be a somewhat common pattern in other cases: consider the
 // intermediate index needed when adding and dropping columns as part of the
 // same transaction.
-func (sb stageBuilder) computeExtraJobOps(s Stage, after []scpb.Status) []scop.Op {
+func (bc buildContext) computeExtraJobOps(s, next *Stage) []scop.Op {
+	revertible := next != nil && next.Phase < scop.PostCommitNonRevertiblePhase
```

Comment:
Yes, this is a much more sane way to approach this. I like this PR even if just for that.